pkg/controller/v1beta1/inferenceservice/reconcilers/hpa/hpa_reconciler.go

```diff
@@ -144,7 +144,8 @@ func (r *HPAReconciler) checkHPAExist(client client.Client) (constants.CheckResu
 }
 
 func semanticHPAEquals(desired, existing *autoscalingv2.HorizontalPodAutoscaler) bool {
-	return equality.Semantic.DeepEqual(desired.Spec, existing.Spec)
+	return equality.Semantic.DeepEqual(desired.Spec, existing.Spec) &&
+		equality.Semantic.DeepEqual(desired.ObjectMeta.Annotations, existing.ObjectMeta.Annotations)
```

Comment:
Annotations can include `last-applied` annotation which can be always different

Reply:
Updated to only check specific annotation instead.

Reply:
Can you help explain what the issue is ? I am not sure how the annotation on HPA matters, as if it is set to `external` then it won't create the HPA.

Reply:
If it's a new ISVC with the annotation `serving.kserve.io/autoscalerClass: "external"`, no HPA will be created. 

However, if it's an existing ISVC where HPA is already created with annotation `serving.kserve.io/autoscalerClass: "hpa"`, our users want to change the annotation to "external" and expect that the existing HPA to be deleted.

Reply:
But I do not think we have logic to delete the existing HPA

Reply:
Oh good catch. I didn't include the change when cherry-picking from another branch.